Performing C SOURCE FILE Test CMAKE_HAVE_LIBC_PTHREAD failed with the following output:
Change Dir: /content/drive/MyDrive/10714/hw4/build/CMakeFiles/CMakeTmp

Run Build Command(s):/usr/bin/make -f Makefile cmTC_ee45c/fast && make[1]: Entering directory '/content/drive/MyDrive/10714/hw4/build/CMakeFiles/CMakeTmp'
/usr/bin/make  -f CMakeFiles/cmTC_ee45c.dir/build.make CMakeFiles/cmTC_ee45c.dir/build
make[2]: Entering directory '/content/drive/MyDrive/10714/hw4/build/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_ee45c.dir/src.c.o
/usr/bin/cc -DCMAKE_HAVE_LIBC_PTHREAD  -fPIC  -o CMakeFiles/cmTC_ee45c.dir/src.c.o -c /content/drive/MyDrive/10714/hw4/build/CMakeFiles/CMakeTmp/src.c
Linking C executable cmTC_ee45c
/usr/local/lib/python3.7/dist-packages/cmake/data/bin/cmake -E cmake_link_script CMakeFiles/cmTC_ee45c.dir/link.txt --verbose=1
/usr/bin/cc -fPIC  -rdynamic CMakeFiles/cmTC_ee45c.dir/src.c.o -o cmTC_ee45c 
CMakeFiles/cmTC_ee45c.dir/src.c.o: In function `main':
src.c:(.text+0x3e): undefined reference to `pthread_create'
src.c:(.text+0x4a): undefined reference to `pthread_detach'
src.c:(.text+0x56): undefined reference to `pthread_cancel'
src.c:(.text+0x67): undefined reference to `pthread_join'
src.c:(.text+0x7b): undefined reference to `pthread_atfork'
collect2: error: ld returned 1 exit status
CMakeFiles/cmTC_ee45c.dir/build.make:98: recipe for target 'cmTC_ee45c' failed
make[2]: *** [cmTC_ee45c] Error 1
make[2]: Leaving directory '/content/drive/MyDrive/10714/hw4/build/CMakeFiles/CMakeTmp'
Makefile:127: recipe for target 'cmTC_ee45c/fast' failed
make[1]: *** [cmTC_ee45c/fast] Error 2
make[1]: Leaving directory '/content/drive/MyDrive/10714/hw4/build/CMakeFiles/CMakeTmp'


Source file was:
#include <pthread.h>

static void* test_func(void* data)
{
  return data;
}

int main(void)
{
  pthread_t thread;
  pthread_create(&thread, NULL, test_func, NULL);
  pthread_detach(thread);
  pthread_cancel(thread);
  pthread_join(thread, NULL);
  pthread_atfork(NULL, NULL, NULL);
  pthread_exit(NULL);

  return 0;
}

Determining if the function pthread_create exists in the pthreads failed with the following output:
Change Dir: /content/drive/MyDrive/10714/hw4/build/CMakeFiles/CMakeTmp

Run Build Command(s):/usr/bin/make -f Makefile cmTC_4d2f3/fast && make[1]: Entering directory '/content/drive/MyDrive/10714/hw4/build/CMakeFiles/CMakeTmp'
/usr/bin/make  -f CMakeFiles/cmTC_4d2f3.dir/build.make CMakeFiles/cmTC_4d2f3.dir/build
make[2]: Entering directory '/content/drive/MyDrive/10714/hw4/build/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_4d2f3.dir/CheckFunctionExists.c.o
/usr/bin/cc   -fPIC -DCHECK_FUNCTION_EXISTS=pthread_create -o CMakeFiles/cmTC_4d2f3.dir/CheckFunctionExists.c.o -c /usr/local/lib/python3.7/dist-packages/cmake/data/share/cmake-3.22/Modules/CheckFunctionExists.c
Linking C executable cmTC_4d2f3
/usr/local/lib/python3.7/dist-packages/cmake/data/bin/cmake -E cmake_link_script CMakeFiles/cmTC_4d2f3.dir/link.txt --verbose=1
/usr/bin/cc -fPIC -DCHECK_FUNCTION_EXISTS=pthread_create -rdynamic CMakeFiles/cmTC_4d2f3.dir/CheckFunctionExists.c.o -o cmTC_4d2f3  -lpthreads 
/usr/bin/ld: cannot find -lpthreads
collect2: error: ld returned 1 exit status
CMakeFiles/cmTC_4d2f3.dir/build.make:98: recipe for target 'cmTC_4d2f3' failed
make[2]: *** [cmTC_4d2f3] Error 1
make[2]: Leaving directory '/content/drive/MyDrive/10714/hw4/build/CMakeFiles/CMakeTmp'
Makefile:127: recipe for target 'cmTC_4d2f3/fast' failed
make[1]: *** [cmTC_4d2f3/fast] Error 2
make[1]: Leaving directory '/content/drive/MyDrive/10714/hw4/build/CMakeFiles/CMakeTmp'



